Border uses palettes 4-6 only (#440)
gbdev/pandocs: src/SGB_Command_Border.md

to have no effect and writes to the same VRAM addresses for both BG and
OBJ ???

> Each tile is stored in 4-bit-per-pixel format consisting of bit planes
> 0 and 1 interleaved by row, followed by bit planes 2 and 3 interleaved
> by row.  In effect, each tile consists of two Game Boy tiles, the first
> to determine bits 0 and 1 (choosing among color 0, 1, 2, or 3 within a
> 4-color subpalette), and the second to determine bits 2 and 3 (choosing
> among colors 0-3, 4-7, 8-11, or 12-15).

Reviewer: ```suggestion
Each tile is stored in 4-bit-per-pixel format consisting of bit planes 0 and 1 interleaved by row, followed by bit planes 2 and 3 interleaved by row.
In effect, each tile consists of two Game Boy tiles, the first to determine bits 0 and 1 (choosing among color 0, 1, 2, or 3 within a 4-color subpalette), and the second to determine bits 2 and 3 (choosing among colors 0-3, 4-7, 8-11, or 12-15).
```

```
 000-6FF  BG Map 32x28 Entries of 16 bits each (1792 bytes)
 700-7FF  Not used, don't care

>  800-85F  BG Palette Data (Palettes 4-6, each 16 colors of 16 bits each)

Reviewer: ```suggestion
 800-85F  BG Palette Data (Palettes 4-6, 16 little-endian RGB555 colors each)
```

the Game Boy window (for example, *Mario's Picross* does this, as does
*WildSnake* to a lesser extent).

> A border designed for a modern (post-2006) widescreen television may use the center 256x176 pixels and leave the top and bottom 24 lines blank.

Reviewer: ```suggestion
A border designed for a modern (post-2006) widescreen television may use the center 256×176 pixels and leave the top and bottom 24 lines blank.
```

Using letterbox allows more tile variety in the portion of the border that a widescreen TV's zoom mode does not cut off.

All borders repeat tiles. Assuming that the blank space for the GB
screen is a blank tile, and the letterbox (if any) is a solid tile, a
border defining all unique tiles would have to define this many tiles:

-   (256\*224-160\*144)/64+1 = 537 tiles in full-screen border

> -   (256\*176-160\*144)/64+2 = 346 tiles in letterboxed border

Reviewer: It would be nice if this could use the same math rendering as #350 (polyfills & all)...

https://github.com/gbdev/pandocs/blob/a4049ecdb0fdfe63a8478e012511cf6b49198014/src/Audio_details.md#L169

...but maybe we can tackle that later.

Reviewer 2: Yes, let's open an Issue about this and merge this as it is

-   (256\*224-160\*144)/64+1 = 537 tiles in fullscreen border
-   (256\*176-160\*144)/64+2 = 346 tiles in widescreen border

> Because the CHR RAM allocated by SGB for border holds only 256 tiles,
> a fullscreen border must repeat at least 281 tiles and a widescreen
> border at least 90.

Reviewer: ```suggestion
Because the CHR RAM allocated by SGB for border holds only 256 tiles, a full-screen border must repeat at least 281 tiles and a widescreen border at least 90.
```
What does "widescreen" mean here, though?

Author: "Widescreen" means a border that is letterboxed for modern widescreen televisions. When the user activates the zoom mode on the television, the television crops the top and bottom of the signal. A game with more "modern" design sensibilities may deliberately not put anything in the top and bottom 24 pixels so that users of modern televisions don't miss anything.

Reviewer: Do you think it'd be appropriate to describe this in a footnote?

Author: Yes. It depends on 1. what syntax to use to make footnotes in this Markdown variant, which is unfamiliar to me, and 2. in how much detail we want to describe [overscan](https://www.nesdev.org/wiki/Overscan).

The Super NES supports 8 background palettes.  The SGB system software
(when run in Mesen-S) has been observed to use background palette 0
for the GB screen, palettes 1, 2, 3, and 7 for the menus, and palettes

> 4, 5, and 6 for the border.   Thus a border can use three 15-color

Reviewer 2: ```suggestion
4, 5, and 6 for the border. Thus a border can use three 15-color
```

> The Super NES supports 8 background palettes.  The SGB system software
> (when run in Mesen-S) has been observed to use background palette 0
> for the GB screen, palettes 1, 2, 3, and 7 for the menus, and palettes
> 4, 5, and 6 for the border. Thus a border can use three 15-color
> palettes.

Reviewer: ```suggestion
The Super NES supports 8 background palettes.
The SGB system software (when run in a LLE such as Mesen-S) has been observed to use background palette 0 for the GB screen, palettes 1, 2, 3, and 7 for the menus, and palettes 4, 5, and 6 for the border.
Thus a border can use three 15-color palettes.
```
